### PR TITLE
QGB Use Ecrecover to validate Ethereum signatures

### PR DESCRIPTION
**Pull request by SweeXordious**
Updates the implementation of the Ethereum signer.

Contributes to https://github.com/celestiaorg/celestia-app/issues/604

### CODE REVIEW

**Comment by SweeXordious** [comment]
reverting after 179 attestation. Converting to draft.

**Comment by evan-forbes** [comment]
we still might want to merge this even if it isn't the (sole) source of the issue described in #604. there's no rush tho, we can wait until we figure it out.

**Comment by SweeXordious** [comment]
The next issue found is that the commitments queried from Celestia-core are not deterministic. Will investigate that separately.
Aside from that, we need the changes we have in this PR. So, we can merge if approved.

File: x/qgb/types/ethereum_signer.go
Comment (SweeXordious) [inline x/qgb/types/ethereum_signer.go:35]:
```suggestion
		return common.Address{}, sdkerrors.Wrap(err, "unmarshalling signature' public key failed")
```

File: x/qgb/types/ethereum_signer.go
Comment (rootulp) [inline x/qgb/types/ethereum_signer.go:44]: Out of curiosity, why is this no longer necessary?

**Review by rootulp — COMMENTED** [review]
For my learning: what was wrong with the previous implementation? 
Related: Do we want a unit test that prevents a future `EthAddressFromSignature` regression?

File: x/qgb/types/ethereum_signer.go
Comment (SweeXordious) [reply x/qgb/types/ethereum_signer.go:44]: https://twitter.com/alexberegszaszi/status/1534461421454606336?t=6hUf4_GjhEuFBVdj_Kqzew&s=19

**Comment by SweeXordious** [comment]
@rootulp Nothing is wrong with the implementation, it was alright. I was suspecting it to be the reason why the relayer was failing. But, turned out to be the commitments.
However, I wanted to be the same as what we are using on the smart contract side:
https://github.com/celestiaorg/quantum-gravity-bridge/blob/37b613843c7a0681fe2a8ae82d3422e1f1cfb311/src/QuantumGravityBridge.sol#L140-L148

File: x/qgb/types/ethereum_signer.go
Comment (rootulp) [reply x/qgb/types/ethereum_signer.go:44]: wow so a bunch of smart contracts include this check even though it's unnecessary

**Comment by rootulp** [comment]
> Nothing is wrong with the implementation, it was alright

Thanks for explaining! So no changes were strictly needed in `x/qgb/types/ethereum_signer_test.go`? On first pass, I thought something about the implementation changed b/c values used in the test were updated.

File: x/qgb/types/ethereum_signer.go
Comment (SweeXordious) [reply x/qgb/types/ethereum_signer.go:44]: yes. And also in our case we were adding this check while it's not needed.

**Comment by SweeXordious** [comment]
@rootulp The different values in tests are because we removed the signatures whos `v` is either `27` or `28`.
The reason is that we no longer need to support that legacy change:
https://ethereum.stackexchange.com/a/113505/65649

**Review by adlerjohn — APPROVED** [review]
utACK

**Review by evan-forbes — APPROVED** [review]
:+1:

**Comment by SweeXordious** [comment]
Will not merge until I get to the bottom of the commitments issue.

**Pull request merged**
